From 296cdab025a18975240d33f18b063306f3062bef Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Mon, 28 Sep 2026 08:40:32 +0100
Subject: [PATCH 2/2] gsi: product trim adjustments

gsi product trims: restore vndk compat symlinks for legacy vendor
partitions, and drop the stock deskclock, calculator and music apps
in favour of the rom replacements.
---
 target/product/gsi/Android.bp      | 40 +++++++++++++++++++++++++++++-
 target/product/handheld_product.mk |  3 ---
 2 files changed, 39 insertions(+), 4 deletions(-)

diff --git a/target/product/gsi/Android.bp b/target/product/gsi/Android.bp
index 1dea323..ffbe79e 100644
--- a/target/product/gsi/Android.bp
+++ b/target/product/gsi/Android.bp
@@ -74,6 +74,44 @@ gsi_symlinks = [
     },
 ]
 
+// compat symlinks for legacy vendor partitions: pre-q vendor ld.config.txt files
+// hardcode /system/${LIB}/vndk-${VER} and vndk-sp-${VER} search paths. point them
+// into the matching vndk apex so those paths keep resolving on old devices.
+vndk_compat_symlinks = [
+    {
+        target: "/apex/com.android.vndk.v28/lib",
+        name: "lib/vndk-28",
+    },
+    {
+        target: "/apex/com.android.vndk.v28/lib64",
+        name: "lib64/vndk-28",
+    },
+    {
+        target: "/apex/com.android.vndk.v28/lib",
+        name: "lib/vndk-sp-28",
+    },
+    {
+        target: "/apex/com.android.vndk.v28/lib64",
+        name: "lib64/vndk-sp-28",
+    },
+    {
+        target: "/apex/com.android.vndk.v29/lib",
+        name: "lib/vndk-29",
+    },
+    {
+        target: "/apex/com.android.vndk.v29/lib64",
+        name: "lib64/vndk-29",
+    },
+    {
+        target: "/apex/com.android.vndk.v29/lib",
+        name: "lib/vndk-sp-29",
+    },
+    {
+        target: "/apex/com.android.vndk.v29/lib64",
+        name: "lib64/vndk-sp-29",
+    },
+]
+
 android_filesystem_defaults {
     name: "android_gsi_defaults",
     defaults: [
@@ -81,7 +119,7 @@ android_filesystem_defaults {
         "system_ext_image_defaults",
         "product_image_defaults",
     ],
-    symlinks: gsi_symlinks,
+    symlinks: gsi_symlinks + vndk_compat_symlinks,
     dirs: ["cache"],
     deps: [
         ///////////////////////////////////////////
diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
index b276cf9..4803e57 100644
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -24,11 +24,8 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/media_product.mk)
 PRODUCT_PACKAGES += \
     AppStore \
     Contacts \
-    DeskClock \
-    ExactCalculator \
     Gallery2 \
     LatinIME \
-    Music \
     preinstalled-packages-platform-handheld-product.xml \
     SettingsIntelligence \
     SpeechServices \
-- 
2.55.0

